### PR TITLE
add load method and clarify a few interfaces

### PR DESCRIPTION
**Pull request by akiezun**

### CODE REVIEW

File: src/main/java/org/broadinstitute/gatk/nativebindings/NativeLibrary.java [hunk 1]
```diff
@@ -1,20 +1,21 @@
 package org.broadinstitute.gatk.nativebindings;
 
-import java.nio.file.Path;
+import java.io.File;
 
 /**
  * Common interface for all native libraries.
+ * The GATK engine will close
```
Comment (droazen) [inline src/main/java/org/broadinstitute/gatk/nativebindings/NativeLibrary.java]: Remove this half-completed sentence

File: src/main/java/org/broadinstitute/gatk/nativebindings/pairhmm/PairHMMNativeArguments.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package org.broadinstitute.gatk.nativebindings.pairhmm;
+
+/**
+ * Struct used to pass arguments to the PairHMM computation.
+ */
+public class PairHMMNativeArguments {
+    /**
+     * The maximum number of threads to use in the PairHMM computation.
+     */
+    int maxNuberOfThreads;
```
Comment (droazen) [inline src/main/java/org/broadinstitute/gatk/nativebindings/pairhmm/PairHMMNativeArguments.java]: `Nuber` -> `Number`

**Comment by lbergelson** [comment]
bump the version to 0.0.2 as well I think

File: src/main/java/org/broadinstitute/gatk/nativebindings/pairhmm/PairHMMNativeArguments.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package org.broadinstitute.gatk.nativebindings.pairhmm;
+
+/**
+ * Struct used to pass arguments to the PairHMM computation.
+ */
+public class PairHMMNativeArguments {
```
Comment (droazen) [inline src/main/java/org/broadinstitute/gatk/nativebindings/pairhmm/PairHMMNativeArguments.java:6]: Should we keep this as a "struct" type class with public mutable fields, or make the fields final and set them via a constructor? Thoughts?

File: src/main/java/org/broadinstitute/gatk/nativebindings/pairhmm/PairHmmNativeBinding.java [hunk 4]
```diff
@@ -4,13 +4,13 @@
 
 /**
  * Provides access to the native kernel of the PairHMM computation.
- * GATK will call {@link #initializeClassFields(Class, Class)} once, the
- * call {@link #computeLikelihoods(int, int, ReadDataHolder[], HaplotypeDataHolder[], double[], int)}
- * multiple times and then call {@link #close()} once at the end.
+ * GATK will call {@link #initialize(Class, Class, PairHMMNativeArguments)} once, the
```
Comment (droazen) [inline src/main/java/org/broadinstitute/gatk/nativebindings/pairhmm/PairHmmNativeBinding.java]: the -> then

**Comment by droazen** [comment]
Review complete, back to @akiezun, merge after addressing comments.

**Comment by droazen** [comment]
Er, how can this branch have conflicts??

File: src/main/java/org/broadinstitute/gatk/nativebindings/NativeLibrary.java (hunk 1 above)
Comment (akiezun) [inline src/main/java/org/broadinstitute/gatk/nativebindings/NativeLibrary.java]: fixed

File: src/main/java/org/broadinstitute/gatk/nativebindings/pairhmm/PairHMMNativeArguments.java (hunk 2 above)
Comment (akiezun) [inline src/main/java/org/broadinstitute/gatk/nativebindings/pairhmm/PairHMMNativeArguments.java]: fixed

File: src/main/java/org/broadinstitute/gatk/nativebindings/pairhmm/PairHMMNativeArguments.java (hunk 3 above)
Comment (akiezun) [inline src/main/java/org/broadinstitute/gatk/nativebindings/pairhmm/PairHMMNativeArguments.java:6]: i think structs are easier when dealing with native code.

File: src/main/java/org/broadinstitute/gatk/nativebindings/pairhmm/PairHmmNativeBinding.java (hunk 4 above)
Comment (akiezun) [inline src/main/java/org/broadinstitute/gatk/nativebindings/pairhmm/PairHmmNativeBinding.java]: fixed

**Comment by akiezun** [comment]
@lbergelson can you publish it to central/sonatype?

**Comment by lbergelson** [comment]
released, just needs to sync to central

**Pull request merged**